dogfood.md

```diff
@@ -0,0 +1,167 @@
+
+## Core Features / User Stories
+This section of the proposal describes the actions that a user can perform in the application and provides some basic constraints.
+
+### A User Logs in
+Users should be able to log in with a social login that is provided by the underlying operating system. For example, On Android these accounts will be exposed through the account picker. On platforms where this isn't provided, we should fall back to an OAuth flow using the system browser.
+
+### Users can manage a self profile
+Users should be able to upload and edit information about themselves in the application.  This will include email, a display name, a profile picture, and a short biography.  As much of this information should be auto filled from the social login when a user first logs in.
+
+### User should be able to see recent memes
+Upon logging in, users should be able to see all of the memes which have been recently posted by other users.  They should be able to view older memes by scrolling through the feed in a list style view.  A meme view will show the meme as an image, the display name of the author of the meme, a summry view which displays the number of comments, the number of likes, the date a meme was posted, and provide an option to "like" a meme or "unlike" a previously liked meme.
+
+### User should be able to see comments on a specific meme
+A meme allows users to comment on it.  Comments are displayed on a timeline from oldest to newest.  With the exception of the comment's author, comments are read only.
+
+### Users should be able to comment on a specific meme.
+A user should be able to comment on a meme.  Users are able to edit and delete their own comments.
+
+### Users should be able to upload memes
+A user should be able to create a meme.  Memes may be created from either a template provided in the application, uploaded from a user's local files, or uploaded from a picture that is taken from the application.  Users are also able to delete memes they have uploaded.  Once a meme is generated, it is not able to be edited.
+
+A meme consists of a picture, a comment displayed aligned to the top of the picture, and a comment displayed aligned with the bottom of the picture.  
+
+### Users can delete their activity
+A user should have the ability to delete all of their comments and meme history from the app.  When a user deletes a meme it will also delete all comments associated with that meme
+
+### Where possible offline usage should be supported
+
+Users should be able to create memes, view cached memes, author comments, and like posts while offline.  Once the device returns online the qued activity should be submitted to the server even if the app is backgrounded.
```

Comment:
@danielpassos @wtrocki The offline experience is a key element of Data Sync.
I've written more about this in the sync proposal
https://github.com/aerogear/proposals/pull/40/files?short_path=6cf137e#diff-ae869d560e5f0b69e70db66dfe11b55cR45